Pinken docs for doc8 🩷

biosafetylvl5 opened the pull request:
PS I am **testing out a new repo template**, please comment what you think of it:

The template:

```
**🔍 Existing Tests:**
- ✅ **Required:** I ran existing tests, and they all pass (e.g., `full_test.sh`)
  - Full test suite (`full_test.sh`) passes
  - Other tests (specify) pass
- ❌ **Not Required**

**🧪 Unit Tests:**
- ✅ **Required:** I added Unit tests for new/modified functionality and they *pass*
- ❌ **Not Required**

**🔗 Integration Tests:**
- ✅ **Required:** I added integration tests for new/modified functionality and they *pass*
- ❌ **Not Required**

**👼 Documentation:**
- ✅ **Required:** I added documentation for new/modified functionality
- ❌ **Not Required**

** 🕸️ Updates to Other Repositories:**
- ✅ **Required:** I made updates to these other repositories:
    -  
- ❌ **Not Required**
```

I removed the "explain reasoning" because in most PRs, it's not filled out.

The template in use:

**🔍 Existing Tests:**
- ❌ **Not Required**

**🧪 Unit Tests:**
- ❌ **Not Required**

**🔗 Integration Tests:**
- ❌ **Not Required**

**👼 Documentation:**
- ❌ **Not Required** - Already in `coding-standards.rst`

** 🕸️ Updates to Other Repositories:**
- ❌ **Not Required**

# 👪 Related Issues
fixes #479 

# 🧪 Testing Instructions
Run doc8 at root:

```
doc8 --max-line-length 120 ./docs/source 
```

# 📖 Summary
I edited the docs to comply with `.rst` standards as enforced by `doc8`.

I tried these "black like" helper programs:
- [rstfmt](https://github.com/dzhu/rstfmt)
- [docstrfmt](https://github.com/LilSpazJoekp/docstrfmt)
- and a few others

None worked well. I wrote [pinkrst](https://github.com/biosafetylvl5/pinkrst/tree/main) to help, hence the name of this PR. 
I am fully convinced no one reads all of the PR descriptions. Let me know if you did :)

![how standards proliferate cartoon from xkcd](https://imgs.xkcd.com/comics/standards.png)

# 💻  Output
``` bash
➜  geoips git:(documentation-pinkening) doc8 --max-line-length 120 ./docs/source 
Scanning...
Validating...
========
Total files scanned = 107
Total files ignored = 0
Total accumulated errors = 0
Detailed error counts:
    - doc8.checks.CheckCarriageReturn = 0
    - doc8.checks.CheckIndentationNoTab = 0
    - doc8.checks.CheckMaxLineLength = 0
    - doc8.checks.CheckNewlineEndOfFile = 0
    - doc8.checks.CheckTrailingWhitespace = 0
    - doc8.checks.CheckValidity = 0
 ```
🙏  👼👼👼👼 we can now ascend to linter heaven 👼👼👼👼 🙏

biosafetylvl5 commented on docs/source/releases/v1_10_0a5.rst:
```diff
@@ -138,26 +138,46 @@ Consolidate both dynamic and static sectors under geoips/plugins/yaml/sectors.
 
 ::
 
-  renamed: geoips/plugins/yaml/sectors_dynamic/tc_1024x1024/tc_1km_1024x1024.yaml -> geoips/plugins/yaml/sectors/dynamic/tc_1024x1024/tc_1km_1024x1024.yaml
```
This seems fine, but not ideal? What do others think? At least it's <120 chars now.

jsolbrig replied:
I wish that the trailing lines were indented or something. I suppose that another option might be to migrate to using lists here and it would likely automatically create the indentation.

biosafetylvl5 commented on docs/source/releases/v1_6_0.rst:
```diff
@@ -56,14 +56,18 @@ installation updates and test script bug fixes*
-    modified: tests/outputs/amsr2.tc_overlay.89pct.imagery_annotated_over_Infrared-Gray/20200518_073601_IO012020_amsr2_gcom-w1_89pct_140kts_98p32_res1p0-cr100-bgInfrared-Gray.png.yaml
-    modified: tests/outputs/amsr2.tc_overlay.89pct.imagery_annotated_over_Visible/20200518_073601_IO012020_amsr2_gcom-w1_89pct_140kts_98p32_res1p0-cr100-bgVisible.png.yaml
-    modified: tests/outputs/amsr2_ocean.tc.windspeed.imagery_clean/20200518_073601_IO012020_amsr2_gcom-w1_windspeed_140kts_85p45_1p0-clean.png.yaml
+    modified:
```
This is an unfortunate wrapping :/ and needs to be fixed.

biosafetylvl5 commented on docs/source/releases/v1_6_3.rst:
```diff
@@ -295,11 +286,14 @@ Add “data_times” to AMSR2 TC YAML metadata outputs
-      modified: tests/outputs/amsr2.tc_overlay.89pct.imagery_annotated_over_Infrared-Gray/20200518_073601_IO012020_amsr2_gcom-w1_89pct_140kts_98p32_res1p0-cr100-bgInfrared-Gray.png.yaml
-      modified: tests/outputs/amsr2.tc_overlay.89pct.imagery_annotated_over_Visible/20200518_073601_IO012020_amsr2_gcom-w1_89pct_140kts_98p32_res1p0-cr100-bgVisible.png.yaml
-
+      modified:
```
Same issue with the wrapping here, see prev comment

biosafetylvl5 commented on docs/source/releases/v1_8_1.rst:
```diff
@@ -190,7 +190,10 @@ because no titles/labels in the unsectored test output).
 ::
 
-    renamed: tests/outputs/ssmis.color89.unprojected_image/20200519.084400.F16.ssmis.color89.self_register.100p00.DMSP.15p0.png -> tests/outputs/ssmis.color91.unprojected_image/20200519.084400.F16.ssmis.color91.self_register.100p00.DMSP.15p0.png
+    renamed:
```
Also needs fixed.

biosafetylvl5 commented on docs/source/releases/v1_9_0.rst:
```diff
@@ -42,36 +42,66 @@ Slight changes due to matplotlib 3.7.0 update
+    modified:
+    tests/outputs/abi.static.Infrared.imagery_annotated/20200918.195020.goes-16.abi.Infrared.goes16.45p56.noaa.10p0.png
+    modified:
+    tests/outputs/abi.static.Visible.imagery_annotated/20200918.195020.goes-16.abi.Visible.goes16.41p12.noaa.10p0.png
```
FIX

biosafetylvl5 commented on docs/source/releases/v1_9_1.rst:
```diff
@@ -223,27 +222,48 @@ Replace GEOIPS BASEDIR/geoips_packages_dir with GEOIPS_PACKAGES_DIR
+        modified:
+tests/outputs/abi.tc.Infrared.imagery_annotated/20200918_195020_AL202020_abi_goes-16_Infrared_110kts_100p00_1p0.png.yaml
+        modified:
+tests/outputs/abi.tc.Visible.imagery_annotated/20200918_195020_AL202020_abi_goes-16_Visible_110kts_100p00_1p0.png.yaml
```
FIX

biosafetylvl5 commented on docs/source/starter/mac_installation.rst:
```diff
@@ -65,9 +65,11 @@ If desired, the GeoIPS environment variables can be added to your
 .. code:: bash
 
     # Mamba for Intel chips
-    curl -L -o conda_install.sh https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-x86_64.sh
```
FIX THIS 💢

biosafetylvl5 commented on docs/source/userguide/plugin_development/algorithm.rst:
```diff
@@ -74,7 +74,8 @@ Copy and paste this code into your algorithm file (feel free to remove the comme
 
     def call(
         xobj, # Xarray dataset holding xarrays
-        variables, # list of required input variables for algorithm. Note: Python lists are ordered, so you can count on your list of variables being in the order in which you define them in your product plugin variables
```
This needs manual reformatting

biosafetylvl5 replied:
done

jsolbrig replied:
Did this require manual reformatting because comments don't get wrapped? Which package is handling the wrapping in embedded code blocks?

biosafetylvl5 replied:
This needed manual reformatting because I wanted the comment on a new line instead of simply wrapped. Right now `pinkrst` just wraps lines that are too long. 99% of the time this isn't a problem, but if it's in a code block that's an issue. 

Our line length maximum (120) leaves enough room for a few indents AND our maximum code length (88) so any code that would be auto-wrapped would have been written for the docs explicitly soooo I'm of the opinion we don't have to worry about it too much.

jsolbrig replied:
Sounds good to me!

biosafetylvl5 reviewed (commented):
Some manual changes needed still

biosafetylvl5 commented on docs/source/releases/v1_9_0.rst:
```diff
@@ -42,36 +42,66 @@ Slight changes due to matplotlib 3.7.0 update
+    modified:
+    tests/outputs/abi.tc.Visible.imagery_annotated/20200918_195020_AL202020_abi_goes-16_Visible_110kts_100p00_1p0.png
+    modified:
+tests/outputs/amsr2.global_overlay.37pct.imagery_annotated_over_Infrared-Gray/20200518.062048.gcom-w1.amsr2.37pct.global.10p06.star.20p0.png
```
need to fix?

biosafetylvl5 commented on docs/source/releases/v1_11_0.rst:
```diff
@@ -241,8 +240,12 @@ cause this update, however, but it seems right now.
-  renamed:    tests/outputs/ssmi.tc.37pct.imagery_clean/20200519_080900_IO012020_ssmi_F15_37pct_110kts_50p65_1p0-clean.png.yaml -> tests/outputs/ssmi.tc.37pct.imagery_clean/20200519_090000_IO012020_ssmi_F15_37pct_110kts_50p65_1p0-clean.png.yaml
+  renamed:
+  tests/outputs/ssmi.tc.37pct.imagery_clean/20200519_080900_IO012020_ssmi_F15_37pct_110kts_50p65_1p0-clean.png ->
+  tests/outputs/ssmi.tc.37pct.imagery_clean/20200519_090000_IO012020_ssmi_F15_37pct_110kts_50p65_1p0-clean.png
```
This is really nice! Makes it easy to see file name changes 😄

jsolbrig replied:
Looking good!

biosafetylvl5 commented:
All manual changes done.

biosafetylvl5 reviewed (commented):
I have gone through all the changes and verified them - would greatly appreciate a few more sets of eyes! 👁️ 👁️ 👁️

jsolbrig commented on docs/source/devguide/build_docs.rst:
```diff
@@ -12,4 +12,4 @@
 Building documentation
-**********************
\ No newline at end of file
+**********************
```
What happened here? The lines look identical. Changed line ending character?

jsolbrig replied:
I thought that github marked removed whitespace with slightly darker pink, but that doesn't seem to be the case anymore. This removed a trailing space.

jsolbrig reviewed (changes requested):
I'm generally good with this. Great job! I have two questions and one thing that we should think about.

Question: If you are using Pinkrst, do we need to add it to pyproject.toml? Is it possible to make Pinkrst run as a formatter within vscode?

Issue: Before I approve this, I'd like to think about one thing. The formatting in the changelogs for the changed/added/deleted files is cumbersome. Is there something that we can do to make it cleaner? Maybe this would go along with updating how we handle the changelogs to limit the number of conflicts that we have to resolve?

biosafetylvl5 commented:
What do y'all think about this? It fixes the line wrapping, and also makes the lines shorter/easier to read.

``` rst
modified::

    example/really/long/ohso/long/justthelongest-notshort-superlong/filepath/that-is-SO-long/CHANGELOG_TEMPLATE.rst

renamed::

    really/long/ohso/long/justthelongest-notshort-superlong/filepath/that-is-SO-long/CHANGELOG_TEMPLATE.rst ->
    shortpath/just-an-example/CHANGELOG_TEMPLATE.rst

    really/long/ohso/long/justthelongest-notshort-superlong/filepath/that-is-SO-long/CHANGELOG_TEMPLATE.rst ->
    really/long/ohso/long/justthelongest-notshort-superlong/diffpath/that-is-SO-long/CHANGELOG_TEMPLATE.rst

deleted::

    really/long/ohso/long/justthelongest-notshort-superlong/filepath/that-is-SO-long/CHANGELOG_TEMPLATE.rst
    example/short/path.file

```

See https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#literal-blocks for rst conventions.

jsolbrig commented:
I like that a lot. Let's see what @mindy thinks since she probably spends more time reading the changelogs than anyone else.
